odbc_fdw.c

```diff
@@ -1289,15 +1339,35 @@ odbcIterateForeignScan(ForeignScanState *node)
 			               &NullablePtr);
 
 			sql_data_type(DataTypePtr, ColumnSizePtr, DecimalDigitsPtr, NullablePtr, &sql_type);
+			if (strcmp("bytea", (char*)sql_type.data) == 0)
+			{
+				conversion = HEX_CONVERSION;
+			}
+			if (strcmp("boolean", (char*)sql_type.data) == 0)
+			{
+				conversion = BOOL_CONVERSION;
+			}
+			else if (strncmp("bit(",(char*)sql_type.data,4)==0 || strncmp("varbit(",(char*)sql_type.data,7)==0)
+			{
+				conversion = BIN_CONVERSION;
+			}
 
 			/* Get the position of the column in the FDW table */
 			for (k=0; k<num_of_table_cols; k++)
 			{
 				if (strcmp(table_columns[k].data, (char *) ColumnName) == 0)
 				{
+					SQLULEN min_size = minimum_buffer_size(DataTypePtr);
+					SQLULEN max_size = 8192;
```

Comment:
Text (longvarchar in odbc parlance). In a text column coming from a PostGIS geometry we were getting a size larger than 1M.

This doesn't limit the data that we can read, since we can read it in chunks. But avoids using 1M chunks for every piece of data.